docs/src/content/docs/guides/pages.mdx

```diff
@@ -0,0 +1,137 @@
+
+Read more in the [“Pages” guide in the Astro docs](https://docs.astro.build/en/basics/astro-pages/).
+
+### Using Starlight’s design in custom pages
+
+To use the Starlight layout in custom pages, wrap your page content with the `<StarlightPage />` component.
+This can be helpful if you are generating content dynamically but still want to use Starlight’s design.
+
+```astro
+---
+// src/pages/custom-page/example.astro
+import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
+import CustomComponent from './CustomComponent.astro';
+---
+
+<StarlightPage frontmatter={{ title: 'My custom page' }}>
+	<p>This is a custom page with a custom component:</p>
+	<CustomComponent />
+</StarlightPage>
+```
+
+#### Props
+
+The `<StarlightPage />` component accepts the following props.
+
+##### `frontmatter` (required)
+
+**type:** `StarlightPageFrontmatter`
+
+Set the [frontmatter properties](/reference/frontmatter/) for this page, similar to frontmatter in Markdown pages.
+The [`title`](/reference/frontmatter/#title-required) property is required and all other properties are optional.
+
+The following properties differ from Markdown frontmatter:
+
+- The [`slug`](/reference/frontmatter/#slug) property is not supported and is automatically set based on the custom page’s URL.
+- The [`editUrl`](/reference/frontmatter/#editurl) option requires a URL to display an edit link.
+- The [`sidebar`](/reference/frontmatter/#sidebar) property is not supported. In Markdown frontmatter, this option allows customization of [autogenerated link groups](/reference/configuration/#sidebar), which is not applicable to pages using the `<StarlightPage />` component.
+
+##### `sidebar`
+
+**type:** `SidebarEntry[] | undefined`  
+**default:** the sidebar generated based on the [global `sidebar` config](/reference/configuration/#sidebar)
+
+Provide a custom site navigation sidebar for this page.
+If not set, the page will use the default global sidebar.
+
+##### `hasSidebar`
+
+**type:** `boolean`  
+**default:** `false` if [`frontmatter.template`](/reference/frontmatter/#template) is `'splash'`, otherwise `true`
+
+Set if the sidebar should be displayed on this page.
+
+##### `headings`
+
+**type:** `{ depth: number; slug: string; text: string }[]`  
+**default:** `[]`
+
+Provide an array of all the headings on this page.
+Starlight will generate the page table of contents from these headings if provided.
+
+##### `dir`
+
+**type:** `'ltr' | 'rtl'`  
+**default:** the writing direction for the current locale
+
+Set the writing direction for this page’s content.
+
+##### `lang`
+
+**type:** `string`  
+**default:** the language of the current locale
+
+Set the BCP-47 language tag for this page’s content, e.g. `en`, `zh-CN`, or `pt-BR`.
+
+##### `isFallback`
+
+**type:** `boolean`  
+**default:** `false`
+
+Indicate if this page is untranslated in the current language and using [fallback content](/guides/i18n/#fallback-content).
```

Comment:
```suggestion
Indicate if this page is using [fallback content](/guides/i18n/#fallback-content) because there is no translation for the current language.
```
I feel like the important part here is that it's using fallback content (and the untranslated bit is secondary). So I might do something like this.